File: backend/app/models/supplier_portal.py
from __future__ import annotations
import uuid
import enum
from sqlalchemy import (
    Column, String, Integer, Numeric, Boolean, Text, Date, DateTime,
    ForeignKey, Enum, UniqueConstraint, JSON,
)
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.db.base import Base, TimestampMixin


# ── Enums ─────────────────────────────────────────────────────────────────────

class SPAccountStatus(str, enum.Enum):
    PENDING   = "PENDING"
    ACTIVE    = "ACTIVE"
    SUSPENDED = "SUSPENDED"
    CLOSED    = "CLOSED"


class SPUserRole(str, enum.Enum):
    OWNER     = "OWNER"
    SALES     = "SALES"
    LOGISTICS = "LOGISTICS"
    FINANCE   = "FINANCE"
    QUALITY   = "QUALITY"
    VIEWER    = "VIEWER"
    CUSTOM    = "CUSTOM"


class SPActivityType(str, enum.Enum):
    LOGIN              = "LOGIN"
    VIEW_PO            = "VIEW_PO"
    ACKNOWLEDGE_PO     = "ACKNOWLEDGE_PO"
    PROPOSE_ETA        = "PROPOSE_ETA"
    UPLOAD_INVOICE     = "UPLOAD_INVOICE"
    UPLOAD_DOC         = "UPLOAD_DOC"
    VIEW_PAYMENT       = "VIEW_PAYMENT"
    UPDATE_PROFILE     = "UPDATE_PROFILE"
    INVITE_USER        = "INVITE_USER"
    DEACTIVATE_USER    = "DEACTIVATE_USER"
    OTHER              = "OTHER"


class SPPOResponseStatus(str, enum.Enum):
    PENDING_RESPONSE      = "PENDING_RESPONSE"
    ACCEPTED              = "ACCEPTED"
    ACCEPTED_REVISED_ETA  = "ACCEPTED_REVISED_ETA"
    PARTIALLY_ACCEPTED    = "PARTIALLY_ACCEPTED"
    REJECTED              = "REJECTED"
    UNDER_REVIEW          = "UNDER_REVIEW"


class SPETAStatus(str, enum.Enum):
    ORIGINAL   = "ORIGINAL"
    REVISED    = "REVISED"
    APPROVED   = "APPROVED"
    REJECTED   = "REJECTED"


class SPDocUploadType(str, enum.Enum):
    INVOICE               = "INVOICE"
    DELIVERY_NOTE         = "DELIVERY_NOTE"
    PACKING_LIST          = "PACKING_LIST"
    BILL_OF_LADING        = "BILL_OF_LADING"
    COA                   = "COA"
    MSDS                  = "MSDS"
    SPECIFICATION         = "SPECIFICATION"
    COMPLIANCE_CERT       = "COMPLIANCE_CERT"
    FOOD_SAFETY_CERT      = "FOOD_SAFETY_CERT"
    HALAL_CERT            = "HALAL_CERT"
    TEST_RESULT           = "TEST_RESULT"
    TAX_DOC               = "TAX_DOC"
    OTHER                 = "OTHER"


class SPDocReviewStatus(str, enum.Enum):
    PENDING    = "PENDING"
    ACCEPTED   = "ACCEPTED"
    REJECTED   = "REJECTED"
    SUPERSEDED = "SUPERSEDED"


class SPPaymentVisibility(str, enum.Enum):
    SUBMITTED          = "SUBMITTED"
    UNDER_REVIEW       = "UNDER_REVIEW"
    MATCHED            = "MATCHED"
    BLOCKED            = "BLOCKED"
    APPROVED_PAYMENT   = "APPROVED_PAYMENT"
    PAID               = "PAID"
    PARTIALLY_PAID     = "PARTIALLY_PAID"
    DISPUTED           = "DISPUTED"


class SPAIAgentType(str, enum.Enum):
    COLLABORATION_MONITOR = "COLLABORATION_MONITOR"
    FRICTION_ASSISTANT    = "FRICTION_ASSISTANT"
    RISK_SIGNAL           = "RISK_SIGNAL"


class SPAIRecStatus(str, enum.Enum):
    PENDING       = "PENDING"
    ACKNOWLEDGED  = "ACKNOWLEDGED"
    ACTIONED      = "ACTIONED"
    DISMISSED     = "DISMISSED"


# ── Models ────────────────────────────────────────────────────────────────────

class SPAccount(Base, TimestampMixin):
    __tablename__ = "sp_accounts"

    id                    = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    supplier_id           = Column(UUID(as_uuid=True), ForeignKey("suppliers.id"), nullable=False, unique=True)
    portal_status         = Column(Enum(SPAccountStatus), nullable=False, default=SPAccountStatus.PENDING)
    default_currency      = Column(String(10), nullable=False, default="KES")
    preferred_language    = Column(String(20), nullable=False, default="en")
    allowed_modules_mask  = Column(JSON, nullable=True)
    primary_contact_name  = Column(String(200), nullable=True)
    primary_contact_email = Column(String(200), nullable=True)
    primary_contact_phone = Column(String(50), nullable=True)
    timezone              = Column(String(60), nullable=False, default="Africa/Nairobi")
    can_view_prices       = Column(Boolean, nullable=False, default=True)
    can_view_payment      = Column(Boolean, nullable=False, default=False)
    notes                 = Column(Text, nullable=True)

    users          = relationship("SPUser", back_populates="account", cascade="all, delete-orphan")
    activity_logs  = relationship("SPActivityLog", back_populates="account", cascade="all, delete-orphan")
    documents      = relationship("SPDocument", back_populates="account", cascade="all, delete-orphan")
    po_responses   = relationship("SPPOResponse", back_populates="account", cascade="all, delete-orphan")
    eta_logs       = relationship("SPETALog", back_populates="account", cascade="all, delete-orphan")


class SPUser(Base, TimestampMixin):
    __tablename__ = "sp_users"
    __table_args__ = (UniqueConstraint("email", name="uq_sp_user_email"),)

    id                 = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    account_id         = Column(UUID(as_uuid=True), ForeignKey("sp_accounts.id"), nullable=False)
    full_name          = Column(String(200), nullable=False)
    email              = Column(String(200), nullable=False)
    phone              = Column(String(50), nullable=True)
    login_identifier   = Column(String(200), nullable=True)
    password_hash      = Column(String(200), nullable=True)
    role_type          = Column(Enum(SPUserRole), nullable=False, default=SPUserRole.VIEWER)
    is_active          = Column(Boolean, nullable=False, default=True)
    last_login_at      = Column(DateTime(timezone=True), nullable=True)
    invited_by         = Column(String(200), nullable=True)
    notes              = Column(Text, nullable=True)

    account            = relationship("SPAccount", back_populates="users")
    activity_logs      = relationship("SPActivityLog", back_populates="user")


class SPPermissionProfile(Base, TimestampMixin):
    __tablename__ = "sp_permission_profiles"
    __table_args__ = (UniqueConstraint("role_type", name="uq_sp_perm_role"),)

    id                        = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    role_type                 = Column(Enum(SPUserRole), nullable=False)
    can_view_po               = Column(Boolean, nullable=False, default=True)
    can_acknowledge_po        = Column(Boolean, nullable=False, default=False)
    can_propose_delivery      = Column(Boolean, nullable=False, default=False)
    can_upload_invoice        = Column(Boolean, nullable=False, default=False)
    can_upload_delivery_docs  = Column(Boolean, nullable=False, default=False)
    can_upload_quality_docs   = Column(Boolean, nullable=False, default=False)
    can_view_payment_status   = Column(Boolean, nullable=False, default=False)
    can_manage_users          = Column(Boolean, nullable=False, default=False)
    can_update_profile        = Column(Boolean, nullable=False, default=False)
    notes                     = Column(Text, nullable=True)


class SPActivityLog(Base):
    __tablename__ = "sp_activity_logs"

    id               = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    account_id       = Column(UUID(as_uuid=True), ForeignKey("sp_accounts.id"), nullable=False)
    user_id          = Column(UUID(as_uuid=True), ForeignKey("sp_users.id"), nullable=True)
    activity_type    = Column(Enum(SPActivityType), nullable=False)
    reference_type   = Column(String(80), nullable=True)
    reference_id     = Column(String(80), nullable=True)
    activity_datetime= Column(DateTime(timezone=True), nullable=False)
    ip_or_session    = Column(String(120), nullable=True)
    notes            = Column(Text, nullable=True)

    account          = relationship("SPAccount", back_populates="activity_logs")
    user             = relationship("SPUser", back_populates="activity_logs")


class SPDocument(Base, TimestampMixin):
    __tablename__ = "sp_documents"

    id                  = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    account_id          = Column(UUID(as_uuid=True), ForeignKey("sp_accounts.id"), nullable=False)
    supplier_id         = Column(UUID(as_uuid=True), ForeignKey("suppliers.id"), nullable=False)
    upload_type         = Column(Enum(SPDocUploadType), nullable=False)
    linked_po_id        = Column(UUID(as_uuid=True), nullable=True)
    linked_grn_id       = Column(UUID(as_uuid=True), nullable=True)
    linked_invoice_id   = Column(UUID(as_uuid=True), nullable=True)
    file_name           = Column(String(300), nullable=False)
    file_path           = Column(Text, nullable=False)
    file_size_kb        = Column(Integer, nullable=True)
    uploaded_by_user_id = Column(UUID(as_uuid=True), ForeignKey("sp_users.id"), nullable=True)
    uploaded_at         = Column(DateTime(timezone=True), nullable=False)
    review_status       = Column(Enum(SPDocReviewStatus), nullable=False, default=SPDocReviewStatus.PENDING)
    reviewed_by         = Column(String(200), nullable=True)
    reviewed_at         = Column(DateTime(timezone=True), nullable=True)
    expiry_date         = Column(Date, nullable=True)
    notes               = Column(Text, nullable=True)

    account             = relationship("SPAccount", back_populates="documents")


class SPPOResponse(Base, TimestampMixin):
    __tablename__ = "sp_po_responses"
    __table_args__ = (UniqueConstraint("account_id", "po_id", name="uq_sp_po_response"),)

    id                = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    account_id        = Column(UUID(as_uuid=True), ForeignKey("sp_accounts.id"), nullable=False)
    po_id             = Column(UUID(as_uuid=True), ForeignKey("purchase_orders.id"), nullable=False)
    responded_by_user = Column(UUID(as_uuid=True), ForeignKey("sp_users.id"), nullable=True)
    response_status   = Column(Enum(SPPOResponseStatus), nullable=False, default=SPPOResponseStatus.PENDING_RESPONSE)
    accepted_qty_json = Column(JSON, nullable=True)
    supplier_comment  = Column(Text, nullable=True)
    rejection_reason  = Column(Text, nullable=True)
    proposed_eta      = Column(Date, nullable=True)
    responded_at      = Column(DateTime(timezone=True), nullable=True)

    account           = relationship("SPAccount", back_populates="po_responses")


class SPETALog(Base, TimestampMixin):
    __tablename__ = "sp_eta_logs"

    id              = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    account_id      = Column(UUID(as_uuid=True), ForeignKey("sp_accounts.id"), nullable=False)
    po_id           = Column(UUID(as_uuid=True), ForeignKey("purchase_orders.id"), nullable=False)
    proposed_by     = Column(UUID(as_uuid=True), ForeignKey("sp_users.id"), nullable=True)
    original_date   = Column(Date, nullable=True)
    proposed_date   = Column(Date, nullable=False)
    eta_status      = Column(Enum(SPETAStatus), nullable=False, default=SPETAStatus.REVISED)
    revision_no     = Column(Integer, nullable=False, default=1)
    supplier_reason = Column(Text, nullable=True)
    buyer_notes     = Column(Text, nullable=True)
    reviewed_by     = Column(String(200), nullable=True)
    reviewed_at     = Column(DateTime(timezone=True), nullable=True)

    account         = relationship("SPAccount", back_populates="eta_logs")


class SPAIRecommendation(Base, TimestampMixin):
    __tablename__ = "sp_ai_recommendations"

    id              = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    account_id      = Column(UUID(as_uuid=True), ForeignKey("sp_accounts.id"), nullable=True)
    agent_type      = Column(Enum(SPAIAgentType), nullable=False)
    status          = Column(Enum(SPAIRecStatus), nullable=False, default=SPAIRecStatus.PENDING)
    title           = Column(String(300), nullable=False)
    body            = Column(Text, nullable=False)
    severity        = Column(String(20), nullable=False, default="INFO")
    reference_type  = Column(String(80), nullable=True)
    reference_id    = Column(String(80), nullable=True)
    actioned_by     = Column(String(200), nullable=True)
    actioned_at     = Column(DateTime(timezone=True), nullable=True)

File: frontend/src/lib/supplier_portal.ts
const BASE = `${process.env.NEXT_PUBLIC_API_URL ?? "http://localhost:8000"}/api/v1/supplier-portal`;

async function r<T>(res: Response): Promise<T> {
  if (!res.ok) throw new Error(await res.text());
  return res.json();
}

// ── Types ─────────────────────────────────────────────────────────────────────

export type SPAccountStatus = "PENDING" | "ACTIVE" | "SUSPENDED" | "CLOSED";
export type SPUserRole = "OWNER" | "SALES" | "LOGISTICS" | "FINANCE" | "QUALITY" | "VIEWER" | "CUSTOM";
export type SPActivityType = "LOGIN" | "VIEW_PO" | "ACKNOWLEDGE_PO" | "PROPOSE_ETA" | "UPLOAD_INVOICE" | "UPLOAD_DOC" | "VIEW_PAYMENT" | "UPDATE_PROFILE" | "INVITE_USER" | "DEACTIVATE_USER" | "OTHER";
export type SPPOResponseStatus = "PENDING_RESPONSE" | "ACCEPTED" | "ACCEPTED_REVISED_ETA" | "PARTIALLY_ACCEPTED" | "REJECTED" | "UNDER_REVIEW";
export type SPETAStatus = "ORIGINAL" | "REVISED" | "APPROVED" | "REJECTED";
export type SPDocUploadType = "INVOICE" | "DELIVERY_NOTE" | "PACKING_LIST" | "BILL_OF_LADING" | "COA" | "MSDS" | "SPECIFICATION" | "COMPLIANCE_CERT" | "FOOD_SAFETY_CERT" | "HALAL_CERT" | "TEST_RESULT" | "TAX_DOC" | "OTHER";
export type SPDocReviewStatus = "PENDING" | "ACCEPTED" | "REJECTED" | "SUPERSEDED";
export type SPAIAgentType = "COLLABORATION_MONITOR" | "FRICTION_ASSISTANT" | "RISK_SIGNAL";
export type SPAIRecStatus = "PENDING" | "ACKNOWLEDGED" | "ACTIONED" | "DISMISSED";

export interface SPAccount {
  id: string;
  supplier_id: string;
  portal_status: SPAccountStatus;
  default_currency: string;
  preferred_language: string;
  primary_contact_name?: string;
  primary_contact_email?: string;
  primary_contact_phone?: string;
  timezone: string;
  can_view_prices: boolean;
  can_view_payment: boolean;
  notes?: string;
  created_at?: string;
}

export interface SPUser {
  id: string;
  account_id: string;
  full_name: string;
  email: string;
  phone?: string;
  role_type: SPUserRole;
  is_active: boolean;
  last_login_at?: string;
  invited_by?: string;
  created_at?: string;
}

export interface SPDocument {
  id: string;
  account_id: string;
  supplier_id: string;
  upload_type: SPDocUploadType;
  linked_po_id?: string;
  linked_grn_id?: string;
  linked_invoice_id?: string;
  file_name: string;
  file_path: string;
  file_size_kb?: number;
  uploaded_at: string;
  review_status: SPDocReviewStatus;
  reviewed_by?: string;
  reviewed_at?: string;
  expiry_date?: string;
  notes?: string;
  created_at?: string;
}

export interface SPPOResponse {
  id: string;
  account_id: string;
  po_id: string;
  response_status: SPPOResponseStatus;
  accepted_qty_json?: Record<string, any>;
  supplier_comment?: string;
  rejection_reason?: string;
  proposed_eta?: string;
  responded_at?: string;
  created_at?: string;
}

export interface SPETALog {
  id: string;
  account_id: string;
  po_id: string;
  proposed_date: string;
  original_date?: string;
  eta_status: SPETAStatus;
  revision_no: number;
  supplier_reason?: string;
  buyer_notes?: string;
  reviewed_by?: string;
  reviewed_at?: string;
  created_at?: string;
}

export interface SPAIRec {
  id: string;
  account_id?: string;
  agent_type: SPAIAgentType;
  status: SPAIRecStatus;
  title: string;
  body: string;
  severity: string;
  reference_type?: string;
  reference_id?: string;
  actioned_by?: string;
  actioned_at?: string;
  created_at?: string;
}

export interface SPActivityLog {
  id: string;
  account_id: string;
  user_id?: string;
  activity_type: SPActivityType;
  reference_type?: string;
  reference_id?: string;
  activity_datetime: string;
  ip_or_session?: string;
  notes?: string;
}

export interface SPDashboard {
  account_id: string;
  supplier_name: string;
  open_pos: number;
  pending_acknowledgment: number;
  uploaded_invoices_30d: number;
  uploaded_docs_30d: number;
  expiring_certs: number;
  recent_activity: SPActivityLog[];
  ai_alerts: SPAIRec[];
}

// ── Color / Label Maps ────────────────────────────────────────────────────────

export const SP_STATUS_COLORS: Record<SPAccountStatus, string> = {
  PENDING: "bg-yellow-100 text-yellow-800",
  ACTIVE: "bg-green-100 text-green-800",
  SUSPENDED: "bg-red-100 text-red-800",
  CLOSED: "bg-gray-100 text-gray-700",
};

export const SP_PO_RESPONSE_COLORS: Record<SPPOResponseStatus, string> = {
  PENDING_RESPONSE: "bg-yellow-100 text-yellow-800",
  ACCEPTED: "bg-green-100 text-green-800",
  ACCEPTED_REVISED_ETA: "bg-blue-100 text-blue-800",
  PARTIALLY_ACCEPTED: "bg-orange-100 text-orange-800",
  REJECTED: "bg-red-100 text-red-800",
  UNDER_REVIEW: "bg-purple-100 text-purple-800",
};

export const SP_DOC_REVIEW_COLORS: Record<SPDocReviewStatus, string> = {
  PENDING: "bg-yellow-100 text-yellow-800",
  ACCEPTED: "bg-green-100 text-green-800",
  REJECTED: "bg-red-100 text-red-800",
  SUPERSEDED: "bg-gray-100 text-gray-700",
};

export const SP_SEVERITY_COLORS: Record<string, string> = {
  INFO: "bg-blue-100 text-blue-800",
  WARNING: "bg-yellow-100 text-yellow-800",
  CRITICAL: "bg-red-100 text-red-800",
};

// ── API Client ────────────────────────────────────────────────────────────────

export const spApi = {
  // Accounts
  listAccounts: (status?: string) =>
    fetch(`${BASE}/accounts${status ? `?status=${status}` : ""}`).then(r<SPAccount[]>),
  getAccount: (id: string) => fetch(`${BASE}/accounts/${id}`).then(r<SPAccount>),
  createAccount: (body: Partial<SPAccount>) =>
    fetch(`${BASE}/accounts`, { method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify(body) }).then(r<SPAccount>),
  updateAccount: (id: string, body: Partial<SPAccount>) =>
    fetch(`${BASE}/accounts/${id}`, { method: "PUT", headers: { "Content-Type": "application/json" }, body: JSON.stringify(body) }).then(r<SPAccount>),

  // Users
  listUsers: (accountId: string) => fetch(`${BASE}/accounts/${accountId}/users`).then(r<SPUser[]>),
  inviteUser: (accountId: string, body: Partial<SPUser>, invitedBy = "internal") =>
    fetch(`${BASE}/accounts/${accountId}/users/invite?invited_by=${encodeURIComponent(invitedBy)}`, {
      method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify(body),
    }).then(r<SPUser>),
  deactivateUser: (userId: string) =>
    fetch(`${BASE}/users/${userId}/deactivate`, { method: "POST" }).then(r<SPUser>),

  // Dashboard
  getDashboard: (accountId: string) => fetch(`${BASE}/accounts/${accountId}/dashboard`).then(r<SPDashboard>),

  // Purchase Orders
  listPOs: (accountId: string, status?: string) =>
    fetch(`${BASE}/accounts/${accountId}/purchase-orders${status ? `?status=${status}` : ""}`).then(r<any[]>),
  getPO: (accountId: string, poId: string) =>
    fetch(`${BASE}/accounts/${accountId}/purchase-orders/${poId}`).then(r<any>),
  acknowledgePO: (accountId: string, poId: string, body: Partial<SPPOResponse>) =>
    fetch(`${BASE}/accounts/${accountId}/purchase-orders/${poId}/acknowledge`, {
      method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify(body),
    }).then(r<SPPOResponse>),

  // ETA
  proposeETA: (accountId: string, body: { po_id: string; proposed_date: string; supplier_reason?: string }) =>
    fetch(`${BASE}/accounts/${accountId}/propose-eta`, {
      method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify(body),
    }).then(r<SPETALog>),
  getETAHistory: (accountId: string, poId: string) =>
    fetch(`${BASE}/accounts/${accountId}/eta-history?po_id=${poId}`).then(r<SPETALog[]>),
  reviewETA: (etaId: string, body: { eta_status: string; reviewed_by: string; buyer_notes?: string }) =>
    fetch(`${BASE}/eta/${etaId}/review`, {
      method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify(body),
    }).then(r<SPETALog>),

  // Documents
  listDocuments: (accountId: string, uploadType?: string, reviewStatus?: string) => {
    const p = new URLSearchParams();
    if (uploadType) p.set("upload_type", uploadType);
    if (reviewStatus) p.set("review_status", reviewStatus);
    const qs = p.toString() ? `?${p}` : "";
    return fetch(`${BASE}/accounts/${accountId}/documents${qs}`).then(r<SPDocument[]>);
  },
  uploadDocument: (accountId: string, body: Partial<SPDocument>) =>
    fetch(`${BASE}/accounts/${accountId}/documents`, {
      method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify(body),
    }).then(r<SPDocument>),
  reviewDocument: (docId: string, body: { review_status: string; reviewed_by: string; notes?: string }) =>
    fetch(`${BASE}/documents/${docId}/review`, {
      method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify(body),
    }).then(r<SPDocument>),
  expiringDocuments: (days = 30) => fetch(`${BASE}/documents/expiring?days=${days}`).then(r<SPDocument[]>),

  // Payment Status
  getPaymentStatus: (accountId: string) =>
    fetch(`${BASE}/accounts/${accountId}/payment-status`).then(r<any[]>),

  // Activity
  listActivity: (accountId: string, limit = 100) =>
    fetch(`${BASE}/accounts/${accountId}/activity?limit=${limit}`).then(r<SPActivityLog[]>),

  // AI
  runAI: () => fetch(`${BASE}/ai/run`, { method: "POST" }).then(r<{ generated: number }>),
  listAIRecs: (accountId?: string) =>
    fetch(`${BASE}/ai/recommendations${accountId ? `?account_id=${accountId}` : ""}`).then(r<SPAIRec[]>),
  ackAIRec: (recId: string, body: { status: string; actioned_by: string }) =>
    fetch(`${BASE}/ai/recommendations/${recId}/ack`, {
      method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify(body),
    }).then(r<SPAIRec>),

  // Auth
  login: (email: string, password: string) =>
    fetch(`${BASE}/auth/login`, {
      method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify({ email, password }),
    }).then(r<any>),
  resetPassword: (email: string, new_password: string) =>
    fetch(`${BASE}/auth/reset-password`, {
      method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify({ email, new_password }),
    }).then(r<{ detail: string }>),

  // Reports
  reportAckTime: () => fetch(`${BASE}/reports/po-acknowledgment-time`).then(r<any[]>),
  reportETA: () => fetch(`${BASE}/reports/eta-revisions`).then(r<any[]>),
  reportAdoption: () => fetch(`${BASE}/reports/adoption`).then(r<any[]>),
  reportNotResponding: () => fetch(`${BASE}/reports/not-responding`).then(r<any[]>),
};
